app/models/merge_request_diff.rb

```diff
@@ -47,6 +47,23 @@ def diffs
     @diffs ||= (load_diffs(st_diffs) || [])
   end
 
+  def diffs_no_whitespace
+    # Update ref for merge request
+    merge_request.fetch_ref
```

Comment:
Is this necessary? This should be updated by `RefreshService.` I would expect that `diffs_no_whitespace` should not cause any writes to the system.